sno/working_copy/postgis.py

```diff
@@ -46,17 +46,38 @@ def adapt_geometry_from_db(g, dbcur):
 
 def adapt_timestamp_from_db(t, dbcur):
     # TODO - revisit timezones.
-    if isinstance(t, str):
-        # This makes postgis timestamps behave more like GPKG ones.
-        return t.replace(" ", "T") + "Z"
-    return t
+    if t is None:
+        return t
+    # This makes postgis timestamps behave more like GPKG ones.
+    return str(t).replace(" ", "T") + "Z"
+
+
+def adapt_to_string(v, dbcur):
+    return str(v) if v is not None else None
 
 
 # See https://github.com/psycopg/psycopg2/blob/master/psycopg/typecast_builtins.c
 TIMESTAMP_OID = 1114
 TIMESTAMP = new_type((TIMESTAMP_OID,), "TIMESTAMP", adapt_timestamp_from_db)
 psycopg2.extensions.register_type(TIMESTAMP)
 
+# We mostly want data out of the DB as strings, just as happens in GPKG.
+# Then we can serialise it using MessagePack.
+# TODO - maybe move adapt-to-string logic to MessagePack?
```

Comment:
feels like it should be a wc-specific function, perhaps with utils/base-classes/mixins/helpers to DRY.